proto/httpcodec/http.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+package httpcodec;
+
+option go_package = "github.com/pokt-network/shannon-sdk/httpcodec";
+
+// HTTPRequest represents an HTTP request to be serialized and sent to a RelayMiner
+// within a RelayRequest payload
+message HTTPRequest {
```

Comment:
Two alterantive options:

1. Rename to `POKTHTTPRequest`
2. IF we compile it to a `types` directory, it'll be easier to discren from the defaults

Reply:
`POKTHTTPRequest` makes sense to me. It's widely used across the off-chain actors.